jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/jobs/cm/AbstractAdminJob.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2003 The Apereo Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *             http://opensource.org/licenses/ecl2
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sakaiproject.component.app.scheduler.jobs.cm;
+
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.quartz.Job;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.SchedulerException;
+import org.sakaiproject.authz.api.SecurityAdvisor;
+import org.sakaiproject.authz.api.SecurityService;
+import org.sakaiproject.tool.api.Session;
+import org.sakaiproject.tool.api.SessionManager;
+
+import java.util.Date;
+import java.util.List;
+
+@Slf4j
+abstract public class AbstractAdminJob implements Job {
+
+    @Setter
+    private SessionManager sessionManager;
+    @Setter
+    private SecurityService securityService;
+    @Setter
+    private String adminUser;
+
+    /**
+     * Child classes should implement their work in this method.
+     *
+     * @param jobExecutionContext
+     * @throws JobExecutionException
+     */
+    protected abstract void executeInternal(JobExecutionContext jobExecutionContext) throws JobExecutionException;
+
+    /**
+     * Switch's the current user to the adminUser and pushes a SecurityAdvisor
+     * onto the security stack to avoid authz issues. Then we invoke
+     * executeInternal, and switch the current user back.
+     *
+     * @param jobExecutionContext
+     * @throws JobExecutionException
+     */
+    public void execute(JobExecutionContext jobExecutionContext) throws JobExecutionException {
+        log.info("{} starting...", this.getClass().getCanonicalName());
+
+        Session sakaiSession = sessionManager.getCurrentSession();
+        String currentUserId = sakaiSession.getUserId();
+
+        sakaiSession.setUserId(adminUser);
+        sakaiSession.setUserEid(adminUser);
```

Comment:
I'm not convinced that a session is needed at all. The SecurityAdvisor can and should be used instead of elevating sessions. We have quartz jobs here locally that do this just fine; no session needed.